cache/disk/disk.go

```diff
@@ -352,7 +352,7 @@ func (c *diskCache) writeAndCloseFile(ctx context.Context, r io.Reader, kind cac
 	var err error
 	var sizeOnDisk int64
 
-	if kind == cache.CAS && c.storageMode != casblob.Identity {
+	if kind == cache.CAS {
```

Comment:
Why isn't this change alone sufficient (and why is the casblob.go refactoring required) ? It's hard to tell in the github UI.

Reply:
The `WriteAndCose` method computed and wrote the CAS blob header ([here](https://github.com/buchgr/bazel-remote/blob/7f72df6ffb0913a753afa8ea3db16a54591c7493/cache/disk/casblob/casblob.go#L530-L555)) even when `Identity` was passed, but `uncompressed` expects the blobs to be plain